### DISCUSSION TITLE
IconButton can't control the size of the icon.

### DISCUSSION THREAD

**Discussion by janckerchen** [chakra-ui/chakra-ui#9074] [General] [answered]
### Description

IconButton can't control the size of the icon, As shown, the svg part is always 20x20, even if p={0}.

<img width="504" alt="image" src="https://github.com/user-attachments/assets/4bd65553-b751-49f6-aad9-ba3fdce3e57a">

### Problem Statement/Justification

Using the following code to resize the icon has no effect.

```jsx

    <IconButton size='lg' p={0}>
      <ZoomIn size={48} absoluteStrokeWidth={true}
    </IconButton>

```

### Proposed Solution or API

The icon fills all the space outside of the padding, and in the extreme case of padding={0}, the icon should automatically fill all the button space.

**Comment by segunadebayo** [comment, marked as answer]
I recommend controlling the icon size from the `IconButton`

```tsx
<IconButton size='lg' _icon={{ boxSize: "48px" }}>
      <ZoomIn absoluteStrokeWidth={true}
 </IconButton>
```
